addons/web/static/src/core/overlay/overlay_container.js

```diff
@@ -9,13 +9,26 @@ export class OverlayContainer extends Component {
 
     setup() {
         this.root = useRef("root");
+        this.overlayIsVisiblePending = new Set();
+        this.overlayIsVisible = useState(new Map());
+        onMounted(() => {
+            for (const overlay of this.overlayIsVisiblePending) {
+                this.overlayIsVisible.set(overlay, this.isVisible(overlay));
+            }
+        });
```

Comment:
I feel like we over-complicate things. We just want `isVisible` to be recomputed when the ref changes. What bout this?
```suggestion
        this.state = useState({ rootEl: null });
        useEffect(
            () => (this.state.rootEl = this.root.el),
            () => [this.root.el]
        );
```